app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,68 @@
+class MarketsController < ApplicationController
+
+  before_action :get_market, only: [:edit, :update]
+
+  def get_market
+    @market = Market.find(params[:id])
+  end
+
+  def index
+    # @markets = Market.all
+    if params[:search]
+      @markets = Market.search(params[:search]).order("created_at DESC")
+    else
+      @markets = Market.all.order('created_at DESC')
+    end
+    render :markets
+  end
+
+  def search
+    @markets = Market.search params[:search]
+    render :search_results
+  end
+
+  def show
+    @market_id = params[:id]
+    @market = Market.find(@market_id)
+  end
+
+  # Add a new market
+  def new
+    @market = Market.new
+  end
+
+  def create
+    @market = Market.create(market_params)
+
+    redirect_to :markets
+  end
+
+  # Update an existing market
+  def update
+    @market.update(market_params)
+
+    redirect_to :markets
+  end
+
+  def name_sort
```

Comment:
You can refactor all of these sort methods into one sort method by using another parameter for the sort that you want to do.

```
 def sort
     if params[:search]
      @markets = Market.search(params[:search]).order(params[:sort])
    else
      @markets = Market.order(params[:sort])
    end
    render :markets
  end
```